libevmasm/SimplificationRules.cpp

```diff
@@ -101,9 +101,14 @@ Rules::Rules()
 		{{Instruction::SDIV, {A, B}}, [=]{ return B.d() == 0 ? 0 : s2u(divWorkaround(u2s(A.d()), u2s(B.d()))); }},
 		{{Instruction::MOD, {A, B}}, [=]{ return B.d() == 0 ? 0 : modWorkaround(A.d(), B.d()); }},
 		{{Instruction::SMOD, {A, B}}, [=]{ return B.d() == 0 ? 0 : s2u(modWorkaround(u2s(A.d()), u2s(B.d()))); }},
-		{{Instruction::EXP, {A, B}}, [=]{ return u256(boost::multiprecision::powm(bigint(A.d()), bigint(B.d()), bigint(1) << 256)); }},
+		{{Instruction::EXP, {A, B}}, [=]{
+			// safety case against compile time bomb
+			if (A.d() >= 2 && B.d() >= 256)
+				return ~u256(0);
```

Comment:
All arithmetic is modulo 2^256 unless otherwise noted.  I can come up with two ways of doing exponentiation under modulo.

1. do the exponentiation in big_int and then take the least significant 256 bits

2. do the exponentiation, taking the least siginificant 256 bits on every intermediate numbers

but actually, these return the same result.